Classes/Domain/AbstractAggregateRoot.php

```diff
@@ -13,35 +13,62 @@
 
 use Neos\Cqrs\Event\EventInterface;
 use Neos\Cqrs\Event\EventTransport;
-use Neos\Cqrs\Event\EventType;
+use Neos\Cqrs\Event\EventTypeService;
 use Neos\Cqrs\Message\MessageMetadata;
 use TYPO3\Flow\Annotations as Flow;
-use TYPO3\Flow\Utility\Arrays;
 
 /**
  * Base class for an aggregate root
  */
 abstract class AbstractAggregateRoot implements AggregateRootInterface
 {
     /**
+     * @var EventTypeService
+     * @Flow\Inject
+     */
+    protected $eventTypeService;
+
+    /**
      * @var string
-     * @Flow\Transient
      */
     protected $aggregateIdentifier;
 
     /**
      * @var string
-     * @Flow\Transient
      */
     protected $aggregateName;
 
     /**
      * @var EventTransport[]
-     * @Flow\Transient
      */
     protected $events = [];
 
     /**
+     * When calling self::recordThatMethod in the contructor
+     * the object is not initialized so event are queued
+     *
+     * @var array
+     */
+    protected $pendingEvents = [];
+
+    /**
+     * @var bool
+     */
+    protected $initialized = false;
+
+    /**
+     * Register event listeners based on annotations
```

Comment:
does the doc comment really fit? Looking at the code, it's about re-recording events which have been recorded during the not-yet-initialized-phase, not about registering event listeners.

And maybe find an even more specific name for `pendingEvents` which makes it clear that these are only events which occurred in the constructor?